### DISCUSSION TITLE
Environment monitor - receiving updates?

### DISCUSSION THREAD

**Discussion by katettudelft** [tesseract-robotics/tesseract_ros#193] [Q&A]
Hello!

My colleagues and I have been running into some issues where the ROS environment monitor is not picking up on our changes.

If I roughly go through the process we're using, would you point out any steps you think we might be missing/doing incorrectly?

This is all in a ROS 1 environment.

- We have a tesseract planning server node (based around twc_motion_planning in your ROS workcell repo) monitoring a namespace.
- Our main application contains a client that will connect to this server (again, based on twc_application)
- We have a `work_frame` link which we apply some transforms to. When it comes to using Tesseract however, it seems unaware of the change to the environment, and produces trajectories based on the original position of the link.
- We can see this in RViz: the RobotModel display shows the transformed link, but the TesseractTrajectory display and the EnvironmentState display for our namespace both show the link in the original position.
- We tried to update the environment from our client by using AddLinkCommand to update the link (and attached joint) with the transform. Our subsequent call to plotToolpath worked, using correct link location. This implies that the local environment was updated.
- However, the tesseract planning server monitoring this namespace still did not receive this update.

We've checked and the updates are being sent out on the topics we expect; likewise we have checked our namespace settings and are pretty confident that they're correct. Do you have any suggestions as to where else to start looking? I'm happy to provide more information if it will help.

Thanks in advance!

**Comment by Levi-Armstrong** [comment]
I am not near a computer, but if you are using the planning server node it should publish its environment out on a different namespace to inspect. If it shows the modified environment then I believe there is something wrong in the environment cache class where it is not detecting the change and clearing the cache of environments. What tags are you on?

**Reply by Levi-Armstrong** [reply]
There is an argument in the planning server launch file that you can enable publishing its environment.

**Reply by Levi-Armstrong** [reply]
@katet-tudelft Did you figure out what the issue was?

**Reply by katettudelft** [reply]
Ah, forgive me for not replying sooner.

My colleague managed to resolve it in the end - I'm not sure of the specifics, but it seems that we were not correctly updating the planning server environment. I've asked him to elaborate further when he can (he's a bit swamped right now).

For the record, I'm not sure which tags we were using, but we built them several months ago. I intend to move to the most current version (0.14?) in our next re-work sprint.
